### PR TITLE
Implemented Density map & Clusters for Cohort + Person map for Profiles

### PR DESCRIPTION
**Pull request by pavgra**

### CODE REVIEW

File: service/src/main/java/org/ohdsi/gisservice/service/CohortService.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @PreAuthorize("hasPermission(#cohortId, 'cohortdefinition', 'get') && hasPermission(#dataSourceKey, 'source', 'access')")
+    public GeoBoundingBox getCohortBounds(Integer cohortId, String dataSourceKey) throws IOException, SQLException {
+
+        DataSourceUnsecuredDTO source = sourceService.getDataSourceDTO(dataSourceKey);
+
+        String sql;
+        try (InputStream is = ClassLoader.class.getResourceAsStream(GET_COHORT_BOUNDS_SQL_PATH)) {
+            sql = IOUtils.toString(is);
```
Comment (wivern) [inline service/src/main/java/org/ohdsi/gisservice/service/CohortService.java]: This method is deprecated

File: service/src/main/java/org/ohdsi/gisservice/service/PersonService.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @PreAuthorize("hasPermission(#personId, #dataSourceKey + ':person', 'get')")
+    public PersonLocationHistory getBoundsHistory(Integer personId, String dataSourceKey) throws IOException {
+
+        DataSourceUnsecuredDTO source = sourceService.getDataSourceDTO(dataSourceKey);
+
+        String sql;
+        try (InputStream is = ClassLoader.class.getResourceAsStream(GET_PERSON_BOUNDS_SQL_PATH)) {
+            sql = IOUtils.toString(is);
```
Comment (wivern) [inline service/src/main/java/org/ohdsi/gisservice/service/PersonService.java]: This method is deprecated

File: service/src/main/java/org/ohdsi/gisservice/service/PersonService.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            sql = SqlRender.renderSql(sql, new String[]{"cdmSchema", "personId"}, new String[]{source.getCdmSchema(), personId.toString()});
+            sql = SqlTranslate.translateSql(sql, source.getType().getOhdsiDB());
+        }
+
+        var locationHistory = new PersonLocationHistory();
+        JdbcTemplate jdbcTemplate = Utils.getJdbcTemplate(source);
+        jdbcTemplate.query(sql, rs -> {
+            var location = new PersonLocation();
```
Comment (wivern) [inline service/src/main/java/org/ohdsi/gisservice/service/PersonService.java:47]: Why not use `RowMapper`?

File: service/src/main/java/org/ohdsi/gisservice/security/sso/SsoProvider.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    private RestTemplate restTemplate;
+
+    public Authentication getAuthentication(String token) {
+
+//        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity request = new HttpEntity<>(resolveRequestHeaders(token));
+        ResponseEntity<JsonNode> response;
+
+        try {
+            response = restTemplate.exchange(userInfoUri, HttpMethod.GET, request, JsonNode.class);
```
Comment (wivern) [inline service/src/main/java/org/ohdsi/gisservice/security/sso/SsoProvider.java:39]: Just to clarify, why `RestTemplate` rather than the Feign?

**Review by wivern — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/ohdsi/gisservice/security/sso/SsoProvider.java (hunk 4 above)
Comment (pavgra) [reply service/src/main/java/org/ohdsi/gisservice/security/sso/SsoProvider.java:39]: The idea is to later move the Sso related classes into shared component and be able to use them from other projects. Therefore would not like to obligate others to use Feign

**Review by wivern — APPROVED** [review]

**Pull request merged**